### DISCUSSION TITLE
Mantine component's customization

### DISCUSSION THREAD

**Discussion by startthecode** [mantinedev/mantine#6018] [Q&A]
**1. How can I customize the Mantine component's CSS like below code?**👇👇👇👇

`/** @type {import('tailwindcss').Config} */
const plugin = require("tailwindcss/plugin");
const defaultTheme = require("tailwindcss/defaultTheme");
module.exports = {
  content: ["./*.{html,js,scss}"],
  theme: {
    plugins: [],
    prefix: "tw-",
    important: true,
    // extends
    extend: {
      screens: {},

      padding: {
        sectionPadding: "55px",
        sectionPaddingMOB: "75px",
        sectionPaddingTAB: "75px",
      },

      margin: {
        sectionmargin: "75px",
      },

      dropShadow: {
        primary_shadow: "0 4px 4px rgba(0, 0, 0, 0.25)",
        secondary_shadow: "10px 10px 25px 0px rgba(0, 0, 0, 0.10)",
      },

      colors: {
        primary_clr: "#007A7F",
        secondary_clr: "#444444",
        tertiary_clr: "#3A3947",
        quartary_clr: "#9A9A9A",
        // bg colors
        primary_BG_clr: "#F0ECE7",
        secondary_BG_clr: "#F5F4F4",
        tertiary_BG_clr: "#FFA800",
        quartary_BG_clr: "#B87A02",
        senary_BG_clr: "#D2AF6F",
        septenary_BG_clr: "#5B5B5B",

        // border colors
        primary_border_clr: "#D9D9D9",
        secondary_border_clr: "#E1E1E1",
        teritary_border_clr: "#DFDFDF",

        // borderClrs
      },

      container: {
        center: true,
        screens: {
          sm: "600px",
          md: "728px",
          lg: "984px",
          xl: "1240px",
          "2xl": "1240px",
        },
      },
      // font fontFamily start
      fontFamily: {
        // centabel: ["centabel", ...defaultTheme.fontFamily.sans],
      },
    },
  },
};`

**and Also like this**

`.btn {
  @apply md:py-[12px] py-[12px] sm:px-[40px] px-[25px] text-[14px] font-medium border-[1px] uppercase rounded-[6px] duration-300 hover:text-white flex items-center  max-w-max w-full break-keep text-secondary_clr ease-linear;
}`

3. Is there a way to set props for the button all at once instead of adding them every time we call the component?"

**Comment by rtivital** [comment]
1. https://mantine.dev/styles/styles-api/
2. https://mantine.dev/theming/default-props/

**Reply by RefreshIdeas** [reply]
I dont know why mantine css is not working with react js(no typescript)

npm - npm install @mantine/core @mantine/hooks 
`<MantineProvider withGlobalStyles={true} withNormalizeCSS={true}>
          <BrowserRouter>
            <AllRoutes />
          </BrowserRouter>
        </MantineProvider>`

**Reply by rtivital** [reply]
https://mantine.dev/getting-started/

**Reply by RefreshIdeas** [reply]
const theme = createTheme({
  fontFamily: "Poppins, sans-serif",
  primaryColor: "green",
  colors: {
    theme1: Array.from({ length: 10 }, (_, index) => "#000"),
    theme2: Array.from({ length: 10 }, (_, index) => "#0f0"),
  },
  components: {
    Button: {
      defaultProps: {
        bg: "theme1",
      },
      Childdrens: {
        defaultProps: {
          color: "#000", 
          bg: "theme1",
        },
      },
    },
  },
});

how to set css for childrens of button see above code
